### DISCUSSION TITLE
YAML schema support

### DISCUSSION THREAD

**Discussion by akselerando** [jy95/docusaurus-json-schema-plugin#471] [Ideas]
Love this plugin!

It would be nice to add YAML schema support

A quick and dirty solution could be to allow YAML data as input, but convert it to JSON behind the scenes.
A more proper solution would be even better ofc

**Comment by jy95** [comment]
Hello @akselerando ,

I'm not against supporting it, just that a big part of this discussion is related to https://github.com/jy95/docusaurus-json-schema-plugin/discussions/258#discussioncomment-8190207 : the parsing library, as you can also have YAML in `$ref` for instance. and must work transparently for end-users.

Currently, it seems that [@apidevtools/json-schema-ref-parser](https://github.com/APIDevTools/json-schema-ref-parser) is the best candidate (in lack of better & maintained one) : would be interesting if someone give a try. In theory, most changes would be in the [entrypoint file](https://github.com/jy95/docusaurus-json-schema-plugin/blob/main/src/theme/JSONSchemaViewer/index.tsx) as typings are almost correct (just `resolverOptions` would need a change to be compliant to new parser options) : 

```ts
// What the user provides
export type Props = {
  /**
   * The JSON schema to use
   */
  schema: unknown,
    /**
   * To customize the ref resolving
   * By default, only inline references will be dereferenced by @stoplight/json-ref-resolver
   */
  resolverOptions?: IResolveOpts
  }
  
 // What inner components use
type InnerViewerProperties = {
     schema: JSONSchema
}
```

Would you or someone else like to give a hand ? Thx in advance
